sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/implementation/ConnectionManager.java

```diff
@@ -0,0 +1,330 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.config.implementation;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.policy.ExponentialBackoff;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.data.appconfiguration.ConfigurationClientBuilder;
+import com.azure.identity.ManagedIdentityCredentialBuilder;
+import com.azure.spring.cloud.config.AppConfigurationCredentialProvider;
+import com.azure.spring.cloud.config.ConfigurationClientBuilderSetup;
+import com.azure.spring.cloud.config.health.AppConfigurationStoreHealth;
+import com.azure.spring.cloud.config.pipline.policies.BaseAppConfigurationPolicy;
+import com.azure.spring.cloud.config.properties.AppConfigurationProviderProperties;
+import com.azure.spring.cloud.config.properties.ConfigStore;
+
+/**
+ * Holds a set of connections to an app configuration store with zero to many geo-replications.
+ */
+public class ConnectionManager {
```

Comment:
What should the responsibility of `ConnectionManager` be?

Reply:
I would like it to be the interface between the library and connecting to App Config, so the library didn't need to know about Geo-Replication or other features, but I don't think it will happen in this PR as it would require re-writing the property source locator.

This in theory could make it easier to move away from the property source model when that comes around.